extras/module-node/README.rst

```diff
@@ -0,0 +1,76 @@
+=====================================
+Node.js-like module loading framework
+=====================================
+
+This directory contains an example module resolution and loading framework and
+``require`` implementation based on the Node.js module system:
+
+* https://nodejs.org/api/modules.html
+
+The application needs only to provide the module resolution and loading logic:
+
+* Add ``duk_module_node.c`` to list of C sources to compile.
+
+* Ensure ``duk_module_node.h`` is in the include path.
+
+* Include the extra header in calling code and initialize the bindings::
+
+      #include "duktape.h"
+      #include "duk_module_node.h"
+
+      /* After initializing the Duktape heap or when creating a new
+       * thread with a new global environment:
+       */
+      duk_push_object(ctx);
+      duk_push_c_function(ctx, cb_resolve_module, DUK_VARARGS);
+      duk_put_prop_string(ctx, -2, "resolve");
+      duk_push_c_function(ctx, cb_load_module, DUK_VARARGS);
+      duk_put_prop_string(ctx, -2, "load");
+      duk_module_node_init(ctx);
+
+  Do not call ``duk_module_node_init()`` more than once for the same global
+  environment.  Doing so will put the module system in an inconsistent state.
+
+* The resolve callback is a Duktape/C function which takes the string passed
+  to `require()` and resolves it to a canonical module ID (for Node.js this is
+  usually the fully resolved filename of the module)::
+
+      duk_ret_t cb_resolve_module(duk_context *ctx) {
+          /*
+           *  Entry stack: [ requested_id parent_id ]
+           */
+
+          const char *requested_id = duk_get_string(ctx, 0);
+          const char *parent_id = duk_get_string(ctx, 1);  /* calling module */
+          const char *resolved_id;
+
+          /* Arrive at the canonical module ID somehow. */
+
+          duk_push_string(ctx, resolved_id);
+          return 1;  /*nrets*/
+      }
+
+  If the module ID cannot be resolved, the resolve callback should throw an
+  error, which will propagate out of the ``require()`` call.  Note also that
+  when the global ``require`` is called, the parent ID is an empty string.
+
+* The load callback is a Duktape/C function which takes the resolved module ID
+  and either returns the Ecmascript source code for the module, or populates
+  ``module.exports`` itself and returns undefined (useful for C modules)::
```

Comment:
Just to be sure: it's also allowed to populate symbols into module.exports and *also* return a string (same as with Duktape 1.x)?

Also: I'm assuming it's allowed to replace `module.exports` in the load() function (as with the updated Duktape 1.x behavior)?